### DISCUSSION TITLE
State not updating in the component

### DISCUSSION THREAD

**Discussion by ImamJanjua** [pmndrs/zustand#2682] [Q&A]
So as u can see I have a test component. Here on layout change the useResponsive will call the useEffect which will set the layout. 
So the setting or layout and related is working but the state coming the store is not updating

https://github.com/user-attachments/assets/8e0ee806-da82-401f-a3af-e26306bf48b9

**Comment by dai-shi** [comment]
Can you create a _minimal_ reproduction with StackBlitz?

**Reply by ImamJanjua** [reply]
https://stackblitz.com/edit/stackblitz-starters-vpscmh?file=app%2Fpage.tsx

**Reply by dai-shi** [reply]
Can you make it much smaller? Just one file without Next.js.

Anyway, let's see if anyone can help you.

**Reply by ImamJanjua** [reply]
@dai-shi the smallest thing i can run is a vite project since u need a web view to see the problem. But I think the repo should be the easiest way to  see the problem

**Reply by dai-shi** [reply]
a vite project with one app file would be great. just remove any lines that are not related to the repro.

**Reply by ImamJanjua** [reply]
https://stackblitz.com/edit/vitejs-vite-cbvg2e?file=src%2FApp.tsx

**Reply by dai-shi** [reply]
Thanks.
Here's the fix: https://stackblitz.com/edit/vitejs-vite-m8wvqo?file=src%2FApp.tsx
